test_tools/log_parser/log_gen.py

```diff
@@ -182,30 +197,67 @@ def mk_cb_payload():
 
     return builder.Output()
 
+def mk_cb_reward(_event_id='event_id_0', _value=1, _pdrop=0):
+    builder = flatbuffers.Builder(0)
+
+    NumericOutcomeStart(builder)
+    NumericOutcomeAddValue(builder, -_value)
+    outcome = NumericOutcomeEnd(builder)
+
+    OutcomeEventStart(builder)
+    OutcomeEventAddValueType(builder, OutcomeValue.numeric)
+    OutcomeEventAddValue(builder, outcome)
+
+    builder.Finish(OutcomeEventEnd(builder))
+    
+    reward_payload = builder.Output()
+
+    builder = flatbuffers.Builder(0)
+    event_id = builder.CreateString(_event_id)
+
+    MetadataStart(builder)
+    MetadataAddId(builder, event_id)
+    MetadataAddClientTimeUtc(builder, mk_timestamp(builder))
+    MetadataAddPayloadType(builder, PayloadType.Outcome)
+    MetadataAddEncoding(builder, EventEncoding.Identity)
+    if _pdrop != 0:
```

Comment:
Why are you making it conditional here and not for decision?

Reply:
here is wrong, forgot to fix. thanks for catching.